### PR TITLE
[GPoS] Support manual selection

### PR DESCRIPTION
**Pull request by mmyyrroonn**

### CODE REVIEW

File: cstrml/staking/src/lib.rs [hunk 1]
```diff
@@ -666,6 +666,12 @@ decl_storage! {
 
         /// The earliest era for which we have a pending, unapplied slash.
         EarliestUnappliedSlash: Option<EraIndex>;
+
+        /// Approved candidates to be validators
+        ApprovedValidators get(fn approved_validators): Vec<T::AccountId>;
+
+        /// Manual Selection
+        ManualSelection get(fn manual_selection): bool = false;
```
Comment (badkk) [inline cstrml/staking/src/lib.rs]: ForceSelection

File: cstrml/staking/src/lib.rs [hunk 2]
```diff
@@ -1477,6 +1483,34 @@ decl_module! {
             ensure_root(origin)?;
             <ErasStakingPayout<T>>::remove(era_index);
         }
+
+        #[weight = 1000]
+        fn add_approved_validators(origin, new_validators: Vec<T::AccountId>) {
```
Comment (badkk) [inline cstrml/staking/src/lib.rs]: add_validator_whitelist

File: cstrml/staking/src/lib.rs [hunk 3]
```diff
@@ -1477,6 +1483,34 @@ decl_module! {
...
+                        current_validators.push(validator);
+                    }
+                }
+            });
+        }
+
+        #[weight = 1000]
+        fn remove_approved_validators(origin, old_validators: Vec<T::AccountId>) {
+            ensure_root(origin)?;
+            let mut new_validators = Self::approved_validators();
```
Comment (badkk) [inline cstrml/staking/src/lib.rs]: Y not just remove it

File: cstrml/staking/src/lib.rs [hunk 4]
```diff
@@ -1477,6 +1483,34 @@ decl_module! {
             ensure_root(origin)?;
             <ErasStakingPayout<T>>::remove(era_index);
         }
+
+        #[weight = 1000]
+        fn add_approved_validators(origin, new_validators: Vec<T::AccountId>) {
+            ensure_root(origin)?;
+            <ApprovedValidators<T>>::mutate( |current_validators| {
```
Comment (badkk) [inline cstrml/staking/src/lib.rs]: Y not just replace it？

**Review by badkk — CHANGES_REQUESTED** [review]

**Review by LowEntropyBody — APPROVED** [review]

**Review by badkk — APPROVED** [review]

**Pull request merged**